### PR TITLE
lock buildpack url to revision

### PR DESCRIPTION
**Pull request by bacongobbler**
closes #3042

also fixes an issue in master when BUILDPACK_URL is present: #3078

### CODE REVIEW

File: builder/image/slugbuilder/builder/build.sh [hunk 1]
```diff
@@ -77,9 +77,21 @@ selected_buildpack=
 if [[ -n "$BUILDPACK_URL" ]]; then
     echo_title "Fetching custom buildpack"
 
+    # FIXME: strip single quotes coming from the builder
+    BUILDPACK_URL=$(echo $BUILDPACK_URL | tr -d "'")
```
Comment (carmstrong) [inline builder/image/slugbuilder/builder/build.sh]: It's a shame we still have to do this even with the changes in #3129...

**Comment by carmstrong** [comment]
Code LGTM if this is what's necessary. I'd like to see some regression tests for `BUILDPACK_URL` in the test changes from #3129.

**Comment by bacongobbler** [comment]
ask and ye shall receive: 26dab04

**Comment by carmstrong** [comment]
> ask and ye shall receive

Great. Code LGTM.

**Comment by mboersma** [comment]
Code LGTM.

**Comment by bacongobbler** [comment]
When using a commit:

```
-----> Fetching custom buildpack
warning: Could not find remote branch 98f37cc to clone.
fatal: Remote branch 98f37cc not found in upstream origin
Unexpected end of command stream
-----> Go app detected
```

I'll silence that but everything else is working now locally. Phew!

**Comment by bacongobbler** [comment]
done

**Comment by amrnt** [comment]
Amen :pray:

**Comment by bacongobbler** [comment]
@amrnt hope you don't mind but I also added functionality to target a commit as well, with inspiration coming from https://github.com/deis/deis/pull/3042#discussion_r24301629. branches, tags and commits all work now. :tada:

**Comment by amrnt** [comment]
:+1: @bacongobbler That's awesome! I'm looking forward for v1.4! :dancer:

**Pull request merged**